x/programs/rust/examples/counter-external/src/lib.rs

```diff
@@ -38,15 +38,20 @@ mod tests {
             .run_step(&owner, &Step::create_program(PROGRAM_PATH))
             .expect("should be able to create this program")
             .id;
-        let counter_external = Param::Id(counter_external);
 
         let counter = simulator
             .run_step(&owner, &Step::create_program(counter_path))
             .expect("should be able to create the counter")
             .id;
         let counter = Param::Id(counter);
 
-        let params = vec![counter_external, counter.clone(), owner_key.clone()];
+        let params = vec![
+            TestContext::default()
+                .with_program_id(counter_external)
```

Comment:
It would be better to do `impl From<Id> for TextContext` where all the other fields get default values.

Reply:
Agree with this design decision. Updated!